chunking_methods/gen_csv_labels.py

```diff
@@ -0,0 +1,177 @@
+"""Generates binary annotations using TweetyNet from weakly labeled audio.
+This file should be run from inside the PyHa directory. It also requires
+WTS_chunking.py to be added to the PyHa directory. 
+Input:     A path to a folder with audio files
+Output:    A csv with chunked, strongly-labeled annotations
+"""
+
+import os
+import sys
+from math import ceil
+import pandas as pd
+from pydub import AudioSegment
+# pylint: disable=import-error #this file gets put into PyHa
+from PyHa.IsoAutio import generate_automated_labels
+from WTS_chunking import dynamic_yan_chunking
+
+ISOLATION_PARAMETERS = {
+    "model" : "tweetynet",
+    "tweety_output": True,
+    "verbose" : True
+}
+
+CHUNK_DURATION = 5
+# incoming filetype
+FILETYPE = ".mp3"
+# use sliding window or raw chunks
+SLIDING_CHUNKS = False
+
+# input weak annotations
+METADATA_PATH = "~/metadata.csv"
+# output for strong annotations
+STRONG_LABELS_CSV = "~/132PeruXC_Labels.csv"
+# output for chunked annotations
+CHUNKED_CSV = "~/132PeruXC_RawChunks.csv"
+
+def convert2wav(path):
+    """Convert audio files to .wav files with PyDub. Used to ensure
+    that TweetyNet can read the files for predictions.
+    Args:
+        path (string)
+            - folder path containing audio files without subfolders
+    """
+    # conversion not needed for tweetynet processing
+    if SLIDING_CHUNKS and FILETYPE in [".wav", ".mp3"]:
+        return
+    for fn in os.listdir(path):
+        if fn.endswith(FILETYPE):
+            x = AudioSegment.from_file(os.path.join(path, fn))
+            x.export(fn.replace(FILETYPE, '.wav'), format='wav')
+
+def generate_labels(path):
+    """Generate binary automated time-specific labels using PyHa
+
+    Args:
+        path (string)
+            - folder path containing audio files with at most one 
+            subdirectory level
+    Returns a PyHa-formatted DataFrame
+    """
+    if not os.path.exists(os.path.join(path)):
+        print(f"Directory not found in path {path}", file=sys.stderr)
+        sys.exit(1)
+
+    # generate labels at a top level
+    convert2wav(path)
+    automated_df = generate_automated_labels(path, ISOLATION_PARAMETERS)
+
+    # check one-level deep in case files organized by class
+    subfolders = [f.path for f in os.scandir(path) if f.is_dir()]
+    if subfolders:
+        subfolders.sort()
+        for s in subfolders:
+            convert2wav(path)
+            temp_df = generate_automated_labels(s, ISOLATION_PARAMETERS)
+            if temp_df.empty:
+                continue
+            automated_df = pd.concat([automated_df, temp_df], ignore_index=True, sort=False)
+    
+    if automated_df.empty:
+        print("no labels generated")
```

Comment:
maybe "cannot automatically determine classes" and add a potential case for finding it from XC filenames?